internal/provider/resource_ko_build.go

```diff
@@ -18,105 +17,239 @@ import (
 			},
-			WorkingDirKey: {
-				Description: "working directory for the build",
-				Optional:    true,
-				Default:     ".",
-				Type:        schema.TypeString,
-				ForceNew:    true, // Any time this changes, don't try to update in-place, just create it.
+			"platforms": schema.ListAttribute{
+				Description:   "platforms to build for",
+				Optional:      true,
+				Computed:      true,
+				ElementType:   basetypes.StringType{},
+				Default:       listdefault.StaticValue(types.ListValueMust(basetypes.StringType{}, []attr.Value{basetypes.NewStringValue("linux/amd64")})),
+				PlanModifiers: []planmodifier.List{listplanmodifier.RequiresReplace()},
+				// TODO: validate platforms here.
 			},
-			PlatformsKey: {
-				Description: "Which platform to use when pulling a multi-platform base. Format: all | <os>[/<arch>[/<variant>]][,platform]*",
-				Optional:    true,
-				Type:        schema.TypeList,
-				Elem:        &schema.Schema{Type: schema.TypeString},
-				ForceNew:    true, // Any time this changes, don't try to update in-place, just create it.
+			"base_image": schema.StringAttribute{
+				Description:   "base image to use",
+				Optional:      true,
+				Computed:      true,
+				Default:       stringdefault.StaticString(defaultBaseImage),
+				PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplace()},
 			},
-			BaseImageKey: {
-				Description: "base image to use",
-				Default:     defaultBaseImage,
-				Optional:    true,
-				Type:        schema.TypeString,
-				ForceNew:    true, // Any time this changes, don't try to update in-place, just create it.
+			"sbom": schema.StringAttribute{
+				Description:   "The SBOM media type to use (none will disable SBOM synthesis and upload, also supports: spdx, cyclonedx, go.version-m).",
+				Optional:      true,
+				Computed:      true,
+				Default:       stringdefault.StaticString("none"),
+				Validators:    []validator.String{sbomValidator{}},
+				PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplace()},
 			},
-			SBOMKey: {
-				Description: "The SBOM media type to use (none will disable SBOM synthesis and upload, also supports: spdx, cyclonedx, go.version-m).",
-				Default:     "spdx",
-				Optional:    true,
-				Type:        schema.TypeString,
-				ForceNew:    true, // Any time this changes, don't try to update in-place, just create it.
-				ValidateDiagFunc: func(data interface{}, _ cty.Path) diag.Diagnostics {
-					v := data.(string)
-					if _, found := validTypes[v]; !found {
-						return diag.Errorf("Invalid sbom type: %q", v)
-					}
-					return nil
-				},
+			"repo": schema.StringAttribute{
+				Description:   "Container repository to publish images to. If set, this overrides the provider's docker_repo, and the image name will be exactly the specified `repo`, without the importpath appended.",
+				Optional:      true,
+				PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplace()},
```

Comment:
Interesting thought exercise...  This is a "ForceNew" but does changing it at the provider level (when it is unspecified here) cause a rebuilt? 🤔 

This might be a "ForceNew" hole in our logic, but worth trying out (not sure if a test is practical)

Reply:
That's a good point. We can't set plan modifiers on fields at the provider level. Changing it (unless it matched the previous value specified on the resource) would result in a new result with a new ID, but I guess the question is whether TF would understand that as a change that should call `Create` vs `Update`. In either case I think we'd be fine, since `Update` does the same thing as `Create`. But this is based only on my shaky understanding of TF semantics, and not any kind of testing.